### DISCUSSION TITLE
Add verification level requirement to repositories to limit who can submit PR(-reviews)

### DISCUSSION THREAD

**Discussion by Andre601** [community/community#6094] [Pull Requests]
Spam is an always present problem for larger repositories on GitHub.
Whenever there is a popular and hight trafficked repository, there will be automated accounts to spam it with issues, discussions, Pull requests and pull request-reviews.

A prime example is [`github/docs`](https://github.com/github/docs). When I [made a PR there](https://github.com/github/docs/pull/10894) to update some docs did it take less than 20 minutes for an account to post a pointless Change request (which doesn't even change a thing and makes me question why this is even doable). Not even 5 minutes after that did someone else post a useless comment.

This is only a small thing right now but still fills my mailbox with useless stuff.

Because of this recent event would I now suggest implementing a setting that allows you to set the requirements for making issues, PRs and (or at the very least) submitting PR reviews.

The idea is as follows:
- Under the `Manage access` tab in the repository setting can you set a default "Verification level" that should be required to do certain actions.
  - `None`: Everyone with an account can contribute.
  - `Verified Email`: The account needs to have a verified Email.
  - `2FA`: The account needs to have a verified Email and 2FA setup.

I know that this could have a negative impact on future contributions towards open source projects, but given how larger projects often suffer from such spam and how people often are willing to proof they want to contribute would I say that it is negotiable on how much this has an impact on things.

2FA isn't difficult to setup, especially with what GitHub offers as options right now, and people may already have such stuff setup. In addition do many spam accounts not really have any 2FA set up, so this could help with blocking them for good.

Of course, this is kind of wishful thinking but there REALLY needs to be something done here!

**Comment by markcellus** [comment]
The most appropriate solution to constant spamming may not necessarily be the one proposed here. But spam is a real big issue and hopefully the team will add more features that better handle that.
